plugins/jobtype/src/azkaban/jobtype/HadoopSecureWrapperUtils.java

```diff
@@ -81,6 +86,67 @@ public static UserGroupInformation createSecurityEnabledProxyUser(String userToP
   }
 
   /**
+   * Sets up the UserGroupInformation proxyUser object so that calling code can do doAs returns null
+   * if the jobProps does not call for a proxyUser
+   * 
+   * @param jobPropsIn
+   * @param tokenFile
+   *          pass tokenFile if known. Pass null if the tokenFile is in the environmental variable
+   *          already.
+   * @param log
+   * @return returns null if no need to run as proxyUser, otherwise returns valid proxyUser that can
+   *         doAs
+   */
+  public static UserGroupInformation setupProxyUser(Properties jobProps,
```

Comment:
this class internally uses UserGroupInformation.createProxyUser, which is a Hadoop static method, which is very hard to test properly.  Suggest skipping for now